### ISSUE TITLE
32bit compile fails,...

### ISSUE THREAD

**Issue by Selur** [libsndfile/libsndfile#666]
using https://github.com/m-ab-s/media-autobuild_suite compilation fails:
```
21:54:16 ┌ sndfile git  ................................. [Updates found]
21:54:16 ├ Running cmake...
21:55:05 ├ Running build...
Likely error (tail of the failed operation logfile):
                 from ../src/common.h:43,
                 from ../src/caf.c:30:
I:/media-autobuild_suite/msys64/mingw32/lib/gcc/i686-w64-mingw32/10.2.0/include/emmintrin.h:143:1: error: inlining failed in call to 'always_inline' '_mm_load_sd': target specific option mismatch
  143 | _mm_load_sd (double const *__P)
      | ^~~~~~~~~~~
In file included from ../src/caf.c:30:
../src/common.h:1025:11: note: called from here
 1025 |    return _mm_cvtsd_si32 (_mm_load_sd (&x)) ;
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```
see: https://github.com/m-ab-s/media-autobuild_suite/issues/1876

**Comment by evpobr** [comment]
Hi @Selur.

Can you check #668?

**Comment by Selur** [comment]
Sadly those changes didn't help, build seems to stop at the same place.
```
09:41:14 ┌ sndfile git  .............................. [Recently updated]
09:41:14 ├ Running cmake...
09:42:01 ├ Running build...
Likely error (tail of the failed operation logfile):
                 from ../src/common.h:43,
                 from ../src/caf.c:30:
I:/media-autobuild_suite/msys64/mingw32/lib/gcc/i686-w64-mingw32/10.2.0/include/emmintrin.h:143:1: error: inlining failed in call to 'always_inline' '_mm_load_sd': target specific option mismatch
  143 | _mm_load_sd (double const *__P)
      | ^~~~~~~~~~~
In file included from ../src/caf.c:30:
../src/common.h:1025:11: note: called from here
 1025 |    return _mm_cvtsd_si32 (_mm_load_sd (&x)) ;
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
build failed. Check I:/media-autobuild_suite/build/sndfile-git/build-32bit/ab-suite.build.log
This is required for other packages, so this script will exit.
09:42:10   Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/ihY3.zip
Copy and paste [logs.zip](https://0x0.st/ihY3.zip) in the GitHub issue.

```

**Comment by evpobr** [comment]
I don't see `-msse2` in logs. Is this correct commit?

**Comment by Selur** [comment]
I think so,.. I deleted the sndfile-git folder to make sure the script checks out the latest version,...
`git log`
shows:

```
$ git log
commit 9d4706df0c71ca93836650865694d91e41fb5a8f (HEAD -> master, origin/master, origin/HEAD)
Author: evpobr <evpobr@gmail.com>
Date:   Wed Dec 9 15:11:07 2020 +0500

    Fix build on platforms without SSE2

    Include math.h to common.h to fix missing lrint() and lrintf().

commit e7f5137b47710cb83ac91a43327c7faed5067168
Author: evpobr <evpobr@gmail.com>
Date:   Mon Dec 7 11:32:15 2020 +0500

    Implement fast SSE2 lrint and lrintf, closes #660

    lrint() and lrintf() are ridiculously slow when libsndfile is built
    using Visual C++.

    On x86 and AMD64 platforms this patch adds SSE2 optimized versions.

    It is safe to assume that non-SSE2 processors are no longer available to
    avoid unnecessary checks.

:...skipping...
commit 9d4706df0c71ca93836650865694d91e41fb5a8f (HEAD -> master, origin/master, origin/HEAD)
Author: evpobr <evpobr@gmail.com>
Date:   Wed Dec 9 15:11:07 2020 +0500

    Fix build on platforms without SSE2

    Include math.h to common.h to fix missing lrint() and lrintf().
```

**Comment by evpobr** [comment]
Tested with MinGW32, i can confirm bug. For me #668 works.
